### PR TITLE
spendingPasswordLastUpdate with empty password, integration test.

### PR DESCRIPTION
**Pull request by denisshevchenko**
<p align="right">#166</p>

# Overview

- [x] I have added integration test for `spendingPasswordLastUpdate` when password was removed.

# Comments

### CODE REVIEW

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 1]
```diff
@@ -808,6 +808,39 @@ spec = do
             , expectFieldEqual assuranceLevel StrictAssurance
             ]
 
+    scenario "WALLETS_UPDATE_04 - updating password updates date in spendingPasswordLastUpdate" $ do
```
Comment (piotr-iohk) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: To be inline with the excel spreadsheet it should be WALLETS_UPDATE_PASS_04 :)

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 2]
```diff
@@ -808,6 +808,39 @@ spec = do
...
+        bkpPhrase <- mkBackupPhrase testBackupPhrase
+        let Right initSpendingPassword = mkSpendingPassword
+                "5416b2988745725998907addf4613c9b0764f04959030e1b81c603b920a115d0"
+        createResp <- successfulRequest $ Client.postWallet $- NewWallet
+            bkpPhrase
+            (Just initSpendingPassword)
+            NormalAssurance
+            "awesomeWallet!"
+            CreateWallet
```
Comment (KtorZ) [inline test/integration/Test/Integration/Scenario/Wallets.hs:81]: This is just:

```hs
            fixture <- setup $ defaultSetup
                & rawPassword .~ "whatever"
```

:+1:

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 3]
```diff
@@ -808,6 +808,39 @@ spec = do
...
+
+        -- 2. Verify that we have a spending password on new wallet.
+        getAgainResp <- request $ Client.getWallet
+            $- (createResp ^. walletId)
+        verify getAgainResp
+            [ expectFieldEqual hasSpendingPassword True
+            ]
+
+        -- 3. Remove spending password.
+        let Right emptySpendingPassword = mkSpendingPassword ""
```
Comment (KtorZ) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: what about `mempty` :) ?

**Review by KtorZ — APPROVED** [review]
Two minor refactors, but otherwise looks good.

**Comment by KtorZ** [comment]
@denis.shevchenko, what about the two refactors suggested :)?
And, I didn't quite spot the lens before... What about just falling back on the field generic lens?

**Comment by denisshevchenko** [comment]
> what about the two reactors suggested :)?

@KtorZ I did one of them (with `mempty` for password). But the second one isn't so easy: we don't have `rawPassword`, so we need to extend `Setup` type. So I thought that it's unnecessary for now...

**Comment by KtorZ** [comment]
@denisshevchenko :upside_down_face: 

https://github.com/input-output-hk/cardano-wallet/blob/develop/test/integration/Test/Integration/Framework/DSL.hs#L169-L174

https://github.com/input-output-hk/cardano-wallet/blob/develop/test/integration/Test/Integration/Framework/DSL.hs#L181-L192

https://github.com/input-output-hk/cardano-wallet/blob/develop/test/integration/Test/Integration/Framework/DSL.hs#L327-L328

**Comment by denisshevchenko** [comment]
Ops, sorry, I forgot to update my `develop`. Done.

**Comment by denisshevchenko** [comment]
Hm, now test fails.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**